src/loaders.py

```diff
@@ -1,5 +1,162 @@
-from .steppy.pytorch.loaders import ImageSegmentationBaseDataset
+import os
+import torch
+from attrdict import AttrDict
+from sklearn.externals import joblib
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.sampler import Sampler
+from PIL import Image
+from steppy.base import BaseTransformer
 
+from .retinanet import DataEncoder
 
-class DetectionLoader(ImageSegmentationBaseDataset):
-    pass
+
+class RandomSubsetSampler(Sampler):
+    def __init__(self, data_size, subset_size):
+        super().__init__()
+        self.data_size = data_size
+        self.subset_size = subset_size
+
+    def __iter__(self):
+        return iter(torch.randperm(self.data_size)[:self.subset_size].long())
+
+    def __len__(self):
+        return self.subset_size
+
+
+class ImageDetectionDataset(Dataset):
+    def __init__(self, X, y, labels, images_dir, encoder, train_mode):
+        super().__init__()
+        self.X = X
+        if y is not None:
+            self.y = y
+            self.labels = labels
+        else:
+            self.y = None
+            self.labels = None
+        self.images_dir = images_dir
+        self.encoder = encoder
+        self.train_mode = train_mode
+
+    def __len__(self):
+        return self.X.shape[0]
+
+    def __getitem__(self, index):
+
+        Xi = self.load_from_disk(index)
+
+        if self.y is not None:
+            yi = self.load_target(index, Xi.size)
+            return Xi, yi
+        else:
+            return Xi
+
+    def load_from_disk(self, index):
+        imgId = self.X[index]
+        img_path = os.path.join(self.images_dir, imgId + '.png')
+        return self.load_from_disk(img_path)
+
+    def load_image(self, img_filepath, grayscale=False):
+        image = Image.open(img_filepath, 'r')
+        if not grayscale:
+            image = image.convert('RGB')
+        else:
+            image = image.convert('L')
+        return image
+
+    def load_target(self, index, img_shape):
+        imgId = self.X[index]
+        boxes_rows = self.y.query('ImageID == {}'.format(imgId))
+        return self.get_boxes_and_labels(boxes_rows, img_shape)
+
+    def get_boxes_and_labels(self, boxes_rows, img_shape):
+        boxes = []
+        labels = []
```

Comment:
@taraspiotr I think that it is better to have it in one line `boxes, labels = [],[]`